pkg/extensions/tbtc/recovery/recovery_test.go

```diff
@@ -102,6 +114,190 @@ func TestBuildSignedTransactionHexString(t *testing.T) {
+		}
+		groupMembers[i] = memberID
+	}
+
+	testData, err := testdata.LoadKeygenTestFixtures(groupSize)
+	if err != nil {
+		t.Fatalf("failed to load test data: [%v]", err)
+	}
+
+	pubKeyToAddressFn := func(publicKey cecdsa.PublicKey) []byte {
+		return elliptic.Marshal(publicKey.Curve, publicKey.X, publicKey.Y)
+	}
+
+	groupMemberAddresses := make([]string, groupSize)
+	for i, member := range groupMembers {
+		pubKey, err := member.PublicKey()
+		if err != nil {
+			t.Fatalf("could not get member pubkey: [%v]", err)
+		}
+		groupMemberAddresses[i] = hex.EncodeToString(pubKeyToAddressFn(*pubKey))
+	}
+
+	errChan := make(chan error)
+
+	waitGroup := &sync.WaitGroup{}
+	waitGroup.Add(groupSize)
+
+	keepAddress := common.Address([20]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1})
+	memberAddresses := make([]common.Address, groupSize)
+	for i, memberAddress := range groupMemberAddresses {
+		memberAddresses[i] = common.HexToAddress(memberAddress)
+	}
+	depositAddressString := "0xa5FA806723A7c7c8523F33c39686f20b52612877"
+	depositAddress := common.HexToAddress(depositAddressString)
+	tbtcHandle.CreateDeposit(depositAddressString, memberAddresses)
+	keep := tbtcHandle.OpenKeep(keepAddress, depositAddress, memberAddresses)
+
+	btcAddresses := []string{
+		"1MjCqoLqMZ6Ru64TTtP16XnpSdiE8Kpgcx",
+		"1EEX8qZnTw1thadyxsueV748v3Y6tTMccc",
+		"1EZuKz6RrJ6XmBPvFwJiEcREpaEVhUVAt5",
+	}
+	maxFeePerVByte := int32(73)
+
+	mutex := &sync.RWMutex{}
+
+	result := make(map[string]string)
+
+	var providersInitializedWg sync.WaitGroup
+	providersInitializedWg.Add(groupSize)
+
+	for i, memberID := range groupMembers {
+		go func(memberID tss.MemberID, index int) {
+			memberPublicKey, err := memberID.PublicKey()
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			memberNetworkKey := key.NetworkPublic(*memberPublicKey)
+			networkProvider := local.ConnectWithKey(&memberNetworkKey)
+
+			providersInitializedWg.Done()
+			providersInitializedWg.Wait()
+
+			defer waitGroup.Done()
+
+			preParams := testData[index].LocalPreParams
+
+			signer, err := tss.GenerateThresholdSigner(
+				ctx,
+				keep.ID().Hex(),
+				memberID,
+				groupMembers,
+				uint(len(groupMembers)-1),
+				networkProvider,
+				pubKeyToAddressFn,
+				params.NewBox(&preParams),
+			)
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			signedHexString, err := BuildBitcoinTransaction(
+				ctx,
+				networkProvider,
+				localChain,
+				tbtcHandle,
+				keep,
+				signer,
+				btcAddresses,
+				maxFeePerVByte,
+			)
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			mutex.Lock()
+			result[memberID.String()] = signedHexString
+			mutex.Unlock()
+		}(memberID, i)
+	}
+
+	go func() {
+		waitGroup.Wait()
+		cancel()
+	}()
+
+	select {
+	case <-ctx.Done():
+		if len(result) != groupSize {
+			t.Errorf(
+				"invalid number of results\nexpected: [%d]\nactual:  [%d]",
+				groupSize,
+				len(result),
+			)
+		}
+
+		expectedSignature := "replace-me"
```

Comment:
`"replace-me"` suggested to me that this is some kind of TODO for later. Let's add a comment explaining what is going to happen once the loop starts.

Reply:
oh whoops - i had a commit that got rid of this approach but it didn't make it in somehow